testsupport/metrics.go

```diff
@@ -0,0 +1,60 @@
+package testsupport
+
+import (
+	"time"
+)
+
+// MetricsAssertionHelper stores baseline metric values when initialized and has convenient functions for metrics assertions
+type MetricsAssertionHelper struct {
+	await          metricsProvider
+	baselineValues map[string]float64
+}
+
+type metricKey string
+
+type metricsProvider interface {
+	GetMetricValue(family string, labels ...string) float64
+	WaitForUserSignupsBeingDeleted(initialDelay time.Duration) error
+	AssertMetricReachesValue(family string, expectedValue float64, labels ...string)
+}
+
+// metric constants
+const (
+	UserSignupsMetric                = "sandbox_user_signups_total"
+	UserSignupsApprovedMetric        = "sandbox_user_signups_approved_total"
+	CurrentMURsMetric                = "sandbox_master_user_record_current"
+	UserSignupsDeactivatedMetric     = "sandbox_user_signups_deactivated_total"
+	UserSignupsAutoDeactivatedMetric = "sandbox_user_signups_auto_deactivated_total"
+	UserSignupsBannedMetric          = "sandbox_user_signups_banned_total"
+)
+
+// InitMetricsAssertion waits for any pending usersignups and then initialized the metrics assertion helper with baseline values
+func InitMetricsAssertion(a metricsProvider) *MetricsAssertionHelper {
+	// Wait for pending usersignup deletions before capturing baseline values so that test assertions are stable
+	a.WaitForUserSignupsBeingDeleted(5 * time.Second)
```

Comment:
this method returns an error that should be handled.

Reply:
Done